### PR TITLE
Added Codewars presence

### PR DESCRIPTION
**Pull request by DragonCat4012**

### CODE REVIEW

**Comment by SuperzZyntexYT** [comment]
I think no one will get this because in most files the class names are included so not mush people will get it.

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
You need to put the logo and thumbnail on imgur instead of Discord's CDN (metadata.json)

**Comment by SlowLife1661** [comment]
And proof of working

File: websites/M/MoodlePresence/presence.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    if (pages[0] === 'grade' && pages[1] === 'report' && pages[2] === 'overview') {
+        data.details = 'Viewing Grade Reports';
+    }
+    if (pages[0] === 'mod' && pages[1] === 'assign' && pages[2].includes('view.php')) {
+        data.details = 'Submitting Assignment';
+        data.state = query(".page-header-headings").textContent;
+    }
+    if (pages[0] === 'login' && pages[1].includes('index.php')) {
+        data.details = 'Logging In . . .';
+    }
```
Comment (SlowLife1661) [inline websites/M/MoodlePresence/presence.ts:55]: else if

File: websites/M/MoodlePresence/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        "en": "Moodle Rich Presence"
+    },
+    "tags": [
+        "moodle"
+    ],
+    "category": "socials",
+    "url": "moodle.com",
+    "logo": "https://cdn.discordapp.com/app-assets/821059536768335902/821099230214750279.png",
+    "thumbnail": "https://cdn.discordapp.com/app-assets/821059536768335902/821099230214750279.png",
+    "regExp": "moodle.(.*?)[.](.*?)"
```
Comment (SlowLife1661) [inline websites/M/MoodlePresence/dist/metadata.json:26]: This regex is wrong

File: websites/M/MoodlePresence/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "description": {
+        "en": "Moodle Rich Presence"
+    },
+    "tags": [
+        "moodle"
+    ],
+    "category": "socials",
+    "url": "moodle.com",
+    "logo": "https://cdn.discordapp.com/app-assets/821059536768335902/821099230214750279.png",
+    "thumbnail": "https://cdn.discordapp.com/app-assets/821059536768335902/821099230214750279.png",
```
Comment (OGMatrix) [inline websites/M/MoodlePresence/dist/metadata.json:25]: Please use Imgur

**Comment by DragonCat4012** [comment]
> And proof of working 

You mean example images? Kk

File: websites/M/MoodlePresence/presence.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+}
+presence.on("UpdateData", async () => {
+    const pathName = document.location.pathname,
+        pages = pathName.split('/').filter(p => p),
+        data: PresenceData = {
+            largeImageKey: "largeimage",
+            smallImageText: 'reading I guess',
+            details: pathName,
+            startTimestamp: timebrowsed
+        };
```
Comment (OGMatrix) [inline websites/M/MoodlePresence/presence.ts:17]: If you only think he does is why using it?

**Comment by DragonCat4012** [comment]
> I think no one will get this because in most files the class names are included so not mush people will get it.

Hmhm Good idea, I will try to make a setting to disable course names

**Review by OGMatrix — CHANGES_REQUESTED** [review]
Please fix it!

**Comment by SlowLife1661** [comment]
Lmao imaging doing mass review I'm out now

File: websites/M/MoodlePresence/dist/metadata.json (hunk 3 above)
Comment (DragonCat4012) [reply websites/M/MoodlePresence/dist/metadata.json:25]: Yeah I already sayed I will fix it later qwq

**Review by Bas950 — CHANGES_REQUESTED** [review]
Upload proof of it working

File: websites/M/MoodlePresence/dist/metadata.json (hunk 2 above)
Comment (DragonCat4012) [reply websites/M/MoodlePresence/dist/metadata.json:26]: huh? why

File: websites/M/MoodlePresence/dist/metadata.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (DragonCat4012) [reply websites/M/MoodlePresence/dist/metadata.json:25]: could I use another image side, which don´t want my phone number to sign up?

File: websites/M/MoodlePresence/presence.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    clientId: "821059536768335902"
+}),
+    timebrowsed = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+    const pathName = document.location.pathname,
+        pages = pathName.split('/').filter(p => p),
+        data: PresenceData = {
+            largeImageKey: "largeimage",
+            smallImageText: 'Reading . . .',
```
Comment (OGMatrix) [inline websites/M/MoodlePresence/presence.ts:11]: Changed "Reading . . ." to "Reading..."

**Review by OGMatrix — CHANGES_REQUESTED** [review]

File: websites/M/MoodlePresence/presence.ts (hunk 5 above)
Comment (DragonCat4012) [reply websites/M/MoodlePresence/presence.ts:11]: why tho

**Comment by DragonCat4012** [comment]
So here are some exmpales qwq

course
dashboard
files
messages

File: websites/M/MoodlePresence/dist/metadata.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (Bas950) [reply websites/M/MoodlePresence/dist/metadata.json:27]: you dont need a phone number for imgur? you dont even need an account

File: websites/M/MoodlePresence/presence.ts (hunk 5 above)
Comment (Bas950) [reply websites/M/MoodlePresence/presence.ts:11]: Dont sert a smallIamgeText if you dont have a smallImage...

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/M/MoodlePresence/dist/metadata.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        "en": "Your end-to-end online learning platform for K12, higher education and workplace training. The most used open source LMS in the world."
+    },
+    "tags": [
+        "moodle",
+        "education",
+        "media"
+    ],
+    "category": "socials",
+    "url": "moodle.com",
+    "logo": "https://imgur.com/a/q2jcE68",
```
Comment (callumok2004) [inline websites/M/MoodlePresence/dist/metadata.json:26]: Use the form "https://i.imgur.com/ABCDEF.png" aka remove "/a", add "i." before imgur.com, and add ".png" at the end

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix codacy issues

**Comment by DragonCat4012** [comment]
forumPNG
kumite
searhingKata
solvingKyu
ViewProfile

File: websites/M/MoodlePresence/dist/metadata.json (hunk 6 above)
Comment (Hans5958) [reply websites/M/MoodlePresence/dist/metadata.json]: Use https://i.imgur.com/ltzuG6b.png

**Comment by Hans5958** [comment]
You also added CodeWars. Please update the PR title.

File: websites/C/Codewars/dist/metadata.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    "author": {
+        "name": " .•` KIΛRΛ `•. ",
+        "id": "586905336850677760"
+    },
+    "contributors": [
+        {
+            "name": "ΛΚΘRΛ",
+            "id": "387655649934311427"
+        }
+    ],
```
Comment (ririxidev) [inline websites/C/Codewars/dist/metadata.json:12]: Delete contributors, didn't see anything that he does in commits, also 
image

File: websites/C/Codewars/dist/metadata.json [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    "tags": [
+        "codewars",
+        "coding",
+        "social"
+    ],
+    "category": "other",
+    "url": "www.codewars.com",
+    "logo": "https://i.imgur.com/BpF8tPx.png",
+    "thumbnail": "https://i.imgur.com/BpF8tPx.png",
```
Comment (ririxidev) [inline websites/C/Codewars/dist/metadata.json:27]: image

File: websites/M/MoodlePresence/dist/metadata.json [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    "author": {
+        "name": " .•` KIΛRΛ `•. ",
+        "id": "586905336850677760"
+    },
+    "contributors": [
+        {
+            "name": "ΛΚΘRΛ",
+            "id": "387655649934311427"
+        }
+    ],
```
[the remark this answers is not in the document]
Comment (ririxidev) [reply websites/M/MoodlePresence/dist/metadata.json:12]: image

File: websites/M/MoodlePresence/dist/metadata.json [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    "tags": [
+        "moodle",
+        "education",
+        "media"
+    ],
+    "category": "socials",
+    "url": "moodle.com",
+    "logo": "https://i.imgur.com/ltzuG6b.png",
+    "thumbnail": "https://i.imgur.com/ltzuG6b.png",
+    "regExp": "moodle.(.*?)[.](.*?)",
```
Comment (ririxidev) [inline websites/M/MoodlePresence/dist/metadata.json:28]: Okay, problem with this regExp is, not every moodle instance is on moodle subdomain. Sometimes moodle is on main domain or even in folder like ``example.com/moodle``. My school have moodle in this way: ``example.com/moodle`` - will not work for me. So i don't really know how you want detect this thing to work everywhere.

**Review by ririxidev — CHANGES_REQUESTED** [review]
You should check out my requested changes:

Also to @callumok2004 @SlowLife1661 @OGMatrix @EncryptedDev, nice that you want help, but you are doing chaos.
image
Please leave reviewing to us.

Thanks,
ririxi (Contact: mail@ririxi.dev, ririxi#2721)

**Comment by ririxidev** [comment]
Hey,

So if I read commits correctly, you are removing RegExp and it will work on main page of moodle, where moodle software isn't a thing? Is that correct? I don't really think that it should be like that, and I don't think that you can detect moodle by something in url so, for me this presence is useless for now and will not be merged. I'm sorry for that, but you should remove this presence from your commits so we can merge your other presence (CodeWars)!

Thanks,
ririxi (Contact: mail@ririxi.dev, ririxi#2721)

File: websites/C/Codewars/dist/metadata.json [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    "url": "www.codewars.com",
+    "logo": "https://i.imgur.com/BpF8tPx.png",
+    "thumbnail": "https://i.imgur.com/Bv7T0JI.png",
+    "settings": [
+        {
+            "id": "statsdisplay",
+            "title": "Dispaly Stats",
+            "value": true
+        }
+    ]
```
Comment (Bas950) [inline websites/C/Codewars/dist/metadata.json:34]: not a valid setting, it is missing some values.

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix codacy issues

**Comment by Bas950** [comment]
Please upload new proof and upload proof of the settings working.

**Comment by DragonCat4012** [comment]
image3
image1
image2

File: websites/C/Codewars/dist/metadata.json [hunk 12]
```diff
@@ -29,7 +29,7 @@
         {
             "id": "statsdisplay",
             "title": "Dispaly Stats",
```
Comment (ririxidev) [inline websites/C/Codewars/dist/metadata.json:31]: "Display" instead of "dispaly"

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @DragonCat4012!
You should check out thing(s) that I requested to change in my review. Please look at them!

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

**Review by Bas950 — APPROVED** [review]

**Review by ririxidev — APPROVED** [review]
LGTM!

**Pull request merged**